photon-ml/build.gradle

```diff
@@ -32,11 +32,11 @@ dependencies {
   }
   compile('org.apache.xmlgraphics:batik-svggen:1.7')
 
-  compile("org.apache.spark:spark-core$scalaSuffix:1.4.0")
-  compile("org.apache.spark:spark-mllib$scalaSuffix:1.4.0")
+  compile("org.apache.spark:spark-core$scalaSuffix:1.6.1")
+  compile("org.apache.spark:spark-mllib$scalaSuffix:1.6.1")
   compile("com.databricks:spark-avro$scalaSuffix:2.0.1")
-  compile("org.apache.spark:spark-sql$scalaSuffix:1.4.0")
-  compile("org.apache.spark:spark-catalyst$scalaSuffix:1.4.0")
+  compile("org.apache.spark:spark-sql$scalaSuffix:1.6.1")
+  compile("org.apache.spark:spark-catalyst$scalaSuffix:1.6.1")
```

Comment:
I haven't tested it for 2.0 yet, but this follows roughly the same progression that spark made from DataFrame in 1.6 toward Dataset in 2.0. This is important to keep in mind, though, and we have a couple other small 2.0 incompatibilities in the code also.